[MIG] product_pricelist_report_qweb: Migrating to v11.0 - T#29976

ebirbe opened the pull request:
Migrating module from Odoo 8.0 to 11.0

![Selección_999(029)](https://user-images.githubusercontent.com/442938/84202938-8a37d880-aa76-11ea-84a2-53b6444cee57.png)

Module `controller_report_xls` is not migrated to 11.0 so this migration only take care of the PDF format version. So, the report format switch was removed from the UI.

Tests were disabled as they are intended only to check the XLS file generation.

There are two fields in the wizard that seems to have no effect in the resulting report (not because of the migrations, it seems to be happening since the original version):

* Only products in pricelist (`only_prod_pricelist`)
* Only products with price (`products_with_price`)

I didn't change the behavior of these two fields until get further information about how they should behave.

/cc @hugho-ad @dsabrinarg

ebirbe committed:
[IMP] product_pricelist_report_qweb: Migrating to v11.0

Migrating module from Odoo 8.0 to 11.0

ebirbe commented:
Dummy PR:

https://git.vauxoo.com/vauxoo/lodigroup/-/merge_requests/1045

luisg123v commented on product_pricelist_report_qweb/__manifest__.py:
```diff
@@ -21,27 +21,23 @@
 {
     "name": "Product Pricelist Report QWeb",
-    "version": "8.0.0.1.6",
+    "version": "11.0.0.1.0",
```
`11.0.1.0.0`

luisg123v commented on product_pricelist_report_qweb/__manifest__.py:
```diff
@@ -21,27 +21,23 @@
-        "view/wizard.xml",
-        "report/layouts.xml",
-        "report/template.xml",
+        "wizard/product_pricelist_views.xml",
```
wizards

luisg123v commented on product_pricelist_report_qweb/__manifest__.py:
```diff
@@ -21,27 +21,23 @@
-        "report/layouts.xml",
-        "report/template.xml",
+        "wizard/product_pricelist_views.xml",
+        "report/product_pricelist_templates.xml",
```
reports

luisg123v commented on product_pricelist_report_qweb/report/product_pricelist.py:
```diff
@@ -0,0 +1,70 @@
+# coding: utf-8
```
Please Remove encoding and license

luisg123v commented on product_pricelist_report_qweb/wizard/product_pricelist.py:
```diff
@@ -19,62 +19,60 @@
 #    You should have received a copy of the GNU Affero General Public License
```
Please remove encoding header and license

luisg123v commented on product_pricelist_report_qweb/report/product_pricelist.py:
```diff
@@ -0,0 +1,70 @@
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+###############################################################################
+from __future__ import division
```
This is python3, there's no need for `__future__`

luisg123v commented on product_pricelist_report_qweb/report/product_pricelist_templates.xml:
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+        <template id="report_pricelist" inherit_id="product.report_pricelist">
```
Wrong indentation, it should be four spaces

luisg123v replied:
Wrong indentation, it should be four spaces

luisg123v commented on product_pricelist_report_qweb/tests/test_report_product.py:
```diff
@@ -41,58 +41,58 @@ class TestReportProductXls(TestXLSProductCommon):
-        self.product.list_price = 0
-        self.columns_number = 10
-        self._generate_report_product()
+    # def test_report_xls_product_pricelist(self):
```
Remove commented code

luisg123v commented on product_pricelist_report_qweb/tests/test_report_product.py:
```diff
@@ -41,58 +41,58 @@ class TestReportProductXls(TestXLSProductCommon):
-            'res_model': 'product.price_list',
-            'res_id': wiz_id.id})
-        self._check_file_xls(attach)
+    # def _generate_report_product(self):
```
Remove commented code

luisg123v commented on product_pricelist_report_qweb/wizard/product_pricelist.py:
```diff
@@ -19,62 +19,60 @@
 #    You should have received a copy of the GNU Affero General Public License
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 ###############################################################################
```
Remove encoding header an license

luisg123v commented on product_pricelist_report_qweb/wizard/product_pricelist.py:
```diff
@@ -19,62 +19,60 @@
 
-class ProductPriceList(osv.osv_memory):
+class ProductPriceList(models.TransientModel):
     _inherit = 'product.price_list'
```
Could you use something less generic, please? ANd without `_` in model name

luisg123v replied:
Comment not applicable, because the model `product.price_list` is native from Odoo, it's just being inherited, so resolving.

luisg123v commented on product_pricelist_report_qweb/wizard/product_pricelist.py:
```diff
@@ -19,62 +19,60 @@
-            'will to add the products that have price.'
-        )
-    }
+    report_format = fields.Selection([
```
`required=True`

luisg123v replied:
This is a model from Odoo, so adding new required fields could cause issues, hence resolving.

luisg123v commented on product_pricelist_report_qweb/wizard/product_pricelist.py:
```diff
@@ -19,62 +19,60 @@
-        context = context and dict(context) or {}
-        ids = isinstance(ids, (int, long)) and [ids] or ids
-        datas = {'ids': context.get('active_ids', [])}
+        if not self.env.user.company_id.logo:
```
```
company = self.env.user.company_id
if not company.***:
```

luisg123v commented on product_pricelist_report_qweb/wizard/product_pricelist_views.xml:
```diff
@@ -1,6 +1,5 @@
-    <data>
+<odoo>
 
         <!--   Product Price List -->
```
Indentation

luisg123v commented on product_pricelist_report_qweb/wizard/product_pricelist_views.xml:
```diff
@@ -9,14 +8,17 @@
-                        <field name="report_format" widget="selection" required='1'/>
-                    </group>
+
+                    <!-- Module `controllers_report_xls` not migrated -->
```
Use word "TODO" to be easily identifiable

sabrinagnu commented:
Functionally 👍 @hugho-ad 
@ebirbe please fix the commented by @luisg123v and let me know to check again.

luisg123v reviewed (changes requested):
Translations need also to be migrated. Could you perform the following, please?
- Re-generate translations and save the result on `es.po`
- Remove file `es_MX.po`

luisg123v commented:
Hi @ebirbe,

Is there any news about this?

luisg123v committed:
[FIX] product_pricelist_report_qweb: Post-migration fixes

The following is performed:
- Removed headers from Python files
- Directories are now in plural (reports, wizards)
- Bumped to correct version (11.0.1.0.0)
- Migrated and updated translations
- Deleted tests, as they have no effect anymore
- And other small changes

luisg123v reviewed (approved):
Given that:
- This has been functionally approved by @dsabrinarg 
- I technically approve this (I just applied small commented suggestiongs: indentation, removing Python headers, bumping to correct version number, etc)
- This is not installed yet in any customer, so it's safe enough
- I'ts been without answer for a long time

I proceed to merge.

CC @moylop260
